tests/dateSelector.spec.ts

```diff
@@ -0,0 +1,113 @@
+    await page.getByText("Jean Coleman").click();
+
+    const samanthaTableStripedSelector = page.locator("table.table-striped").last().locator("tr td dd");
+    const samanthaAddVisitButton = page.locator("table.table-striped").last().getByRole("button", { name: "Add Visit" });
+    await expect(samanthaTableStripedSelector.first()).toHaveText("Samantha");
+    await samanthaAddVisitButton.click();
+
+    await expect(page.locator("h2")).toHaveText("New Visit");
+
+    const newVisitTableStripedSelector = page.locator("table.table-striped").locator("tr td");
+    await expect(newVisitTableStripedSelector.first()).toHaveText("Samantha");
+    await expect(newVisitTableStripedSelector.last()).toHaveText("Jean Coleman");
+
+    const calendarIcon = page.getByLabel("Open calendar");
+    await calendarIcon.click();
+    let date = new Date();
+    const currentYear = date.getFullYear();
+    const currentMonth = date.getMonth() + 1;
+    const currentDay = date.getDate();
+    const currentDate = `${currentYear}/${currentMonth}/${currentDay}`;
+
+    const calendarTodayCell = page.locator('[class="mat-calendar-body-cell-content mat-focus-indicator mat-calendar-body-today"]');
+    await calendarTodayCell.click();
+
+    const dateInputField = await page.locator('input[name="date"]');
+    await expect(dateInputField).toHaveValue(currentDate);
+
+    const descriptionInputField = await page.locator("#description");
+    await descriptionInputField.fill("dermatologist visit");
+    const newVisitAddVisitButton = await page.getByRole("button", {name: "Add Visit"});
+    await newVisitAddVisitButton.click();
+
+    const dermatologistVisitDate = `${currentYear}-${currentMonth}-${currentDay}`;
+    const petVisitsRow = page.locator("table.table-condensed").last().locator("tr td");
+    await expect(petVisitsRow.first()).toHaveText(dermatologistVisitDate);
+
+    await samanthaAddVisitButton.click();
+
+    await calendarIcon.click();
+    let targetDate = new Date();
+    targetDate.setDate(targetDate.getDate() - 45);
+    const secondVisitYear = date.getFullYear();
+    const secondVisitMonth = date.getMonth() + 1;
+    const secondVisitDay = date.getDate();
+    const secondVisitDate = `${secondVisitYear}/${secondVisitMonth}/${secondVisitDay}`;
+
+    await calendarTodayCell.click();
+    await expect(dateInputField).toHaveValue(secondVisitDate);
+
+    await descriptionInputField.fill("massage therapy");
+    await newVisitAddVisitButton.click();
+
+    const massageTherapyVisitDate = `${secondVisitYear}-${secondVisitMonth}-${secondVisitDay}`;
+
+    const samanthaTableCondensedSelector = page.locator("table.table-condensed").last().locator("tr");
+    const firstDate = Date.parse(await samanthaTableCondensedSelector.nth(1).getByRole("cell").first().innerText());
+    const secondDate = Date.parse(await samanthaTableCondensedSelector.nth(2).getByRole("cell").first().innerText());
+    await expect(firstDate).toBeGreaterThan(secondDate);
```

Comment:
This assertion actually fails, because two dates are the same:
![image](https://github.com/user-attachments/assets/3490f73d-4407-48ea-a664-58133531d047)

For some reason, when test runs faster, it picks up 2nd and 3rd rows of the table. instead of first and second, and it's the reason why it pass, because 3rd row date is a lot in the past. 

If you delete all rows with visits for Samantha and run test again, you will see how it's going to fail

Also await is not needed, because `toBeGreaterThan()` is not a locator assertion. By the way, good approach to use this locator compared to the approach suggested in test case notes